### PR TITLE
[Admin] Add payment request index and show pages

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A
|-----------------|-----
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | n/a
| License         | MIT

image
image

### CODE REVIEW

File: src/Sylius/Bundle/AdminBundle/Resources/config/grids/payment_request_payment.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                    label: sylius.ui.action
+                    form_options:
+                        choices:
+                            sylius.ui.authorize: authorize
+                            sylius.ui.cancel: cancel
+                            sylius.ui.capture: capture
+                            sylius.ui.payout: payout
+                            sylius.ui.refund: refund
+                            sylius.ui.status: status
+                            sylius.ui.sync: sync
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/AdminBundle/Resources/config/grids/payment_request_payment.yml]: Use constants for the values, same with states

File: src/Sylius/Bundle/AdminBundle/Resources/config/grids/payment_request_payment.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
+sylius_grid:
+    grids:
+        sylius_admin_payment_payment_request:
+            driver:
+                name: doctrine/orm
+                options:
+                    class: "%sylius.model.payment_request.class%"
+                    repository:
+                        method: createForPaymentQueryBuilder
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/AdminBundle/Resources/config/grids/payment_request_payment.yml]:
```suggestion
                        method: createQueryBuilderForPayment
```

File: src/Sylius/Bundle/AdminBundle/Resources/config/grids/payment_request_payment.yml [hunk 3]
```diff
@@ -0,0 +1,93 @@
+sylius_grid:
+    grids:
+        sylius_admin_payment_payment_request:
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/AdminBundle/Resources/config/grids/payment_request_payment.yml]:
```suggestion
        sylius_admin_payment_request:
```

File: src/Sylius/Bundle/AdminBundle/Resources/config/routing/payment.yml [hunk 4]
```diff
@@ -27,3 +27,29 @@ sylius_admin_payment_complete:
                 transition: complete
             redirect: sylius_admin_payment_index
             flash: sylius.payment.completed
+
+sylius_admin_payment_payment_request_index:
+    path: /payments/{id}/payment_requests
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/AdminBundle/Resources/config/routing/payment.yml]: Both of these routes can be extracted into a separate file and prefixed w/ `/payment/{id}/payment-requests/`

File: src/Sylius/Bundle/AdminBundle/templates/payment/payment_request/show.html.twig [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{% extends '@SyliusAdmin/shared/crud/show.html.twig' %}
+
+{% set resource_name = 'payment.payment_request' %}
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/AdminBundle/templates/payment/payment_request/show.html.twig]: What's with the prefix?

File: src/Sylius/Bundle/CoreBundle/PaymentRequest/Announcer/PaymentRequestAnnouncer.php [hunk 6]
```diff
@@ -13,6 +13,8 @@
 
 namespace Sylius\Bundle\CoreBundle\PaymentRequest\Announcer;
 
+use Sylius\Bundle\CoreBundle\Command\Admin\Account\RequestResetPasswordEmail;
+use Sylius\Bundle\CoreBundle\Command\CapturePaymentRequest;
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/CoreBundle/PaymentRequest/Announcer/PaymentRequestAnnouncer.php]:
```suggestion
```

File: src/Sylius/Bundle/PaymentBundle/Doctrine/ORM/PaymentRequestRepository.php [hunk 7]
```diff
@@ -13,9 +13,11 @@
 
 namespace Sylius\Bundle\PaymentBundle\Doctrine\ORM;
 
+use Doctrine\ORM\QueryBuilder;
 use Sylius\Bundle\ResourceBundle\Doctrine\ORM\EntityRepository;
 use Sylius\Component\Payment\Model\PaymentRequestInterface;
 use Sylius\Component\Payment\Repository\PaymentRequestRepositoryInterface;
+use Symfony\Bridge\Doctrine\Types\UuidType;
```
Comment (NoResponseMate) [inline src/Sylius/Bundle/PaymentBundle/Doctrine/ORM/PaymentRequestRepository.php:20]: Is this package included in the PaymentBundle's composer?

File: features/admin/payment/managing_payments/payment_request/browsing_payment_requests.feature [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+        And there is an "#00000001" order with "PHP T-Shirt" product
+        And the payment request action "authorize" has been executed for order "#00000001" with the payment method "Cash on Delivery"
+        And the payment request action "capture" has been executed for order "#00000001" with the payment method "Cash on Delivery"
+        And the payment request action "sync" has been executed for order "#00000001" with the payment method "Cash on Delivery"
+        And I am logged in as an administrator
+
+    @api @ui
+    Scenario: Browsing payment requests of a payment
+        When I browse payments
+        And I want to view the payment request of the first payment
```
Comment (GSadee) [inline features/admin/payment/managing_payments/payment_request/browsing_payment_requests.feature]:
```suggestion
        And I want to view the payment requests of the first payment
```
🤔

File: features/admin/payment/managing_payments/payment_request/seeing_payment_request_details.feature [hunk 9]
```diff
@@ -0,0 +1,23 @@
+@managing_payments
+Feature: Seeing payment request details
```
Comment (GSadee) [inline features/admin/payment/managing_payments/payment_request/seeing_payment_request_details.feature]:
```suggestion
Feature: Seeing payment request's details
```

File: src/Sylius/Behat/Context/Setup/PaymentContext.php [hunk 10]
```diff
@@ -129,15 +127,15 @@ public function itHasInstructions(PaymentMethodInterface $paymentMethod, $instru
     /**
      * @Given the store has :paymentMethodName payment method not assigned to any channel
      */
-    public function theStoreHasPaymentMethodNotAssignedToAnyChannel($paymentMethodName)
+    public function theStoreHasPaymentMethodNotAssignedToAnyChannel($paymentMethodName): void
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Setup/PaymentContext.php]: You could add also a typehint 😅

File: src/Sylius/Behat/Context/Ui/Admin/ManagingPaymentRequestsContext.php [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+
+    /**
+     * @Then /^there should be (\d+) payment requests? on the list$/
+     */
+    public function thereShouldBeProductVariantsOnTheList(int $count): void
+    {
+        Assert::same(
+            $this->indexPage->countItems(),
+            $count,
+        );
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingPaymentRequestsContext.php]:
```suggestion
        Assert::same($this->indexPage->countItems(), $count);
```

File: src/Sylius/Behat/Page/Admin/Payment/PaymentRequest/IndexPageInterface.php [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+use Sylius\Behat\Page\Admin\Crud\IndexPageInterface as BaseIndexPageInterface;
+
+interface IndexPageInterface extends BaseIndexPageInterface
+{
+    public function choosePaymentMethodToFilter(string $paymentMethodName): void;
+
+    public function chooseActionToFilter(string $action): void;
+
+    public function chooseStateToFilter(string $state): void;
+
```
Comment (GSadee) [inline src/Sylius/Behat/Page/Admin/Payment/PaymentRequest/IndexPageInterface.php]:
```suggestion
```

File: src/Sylius/Bundle/AdminBundle/Resources/config/app/twig_hooks/payment_request/show.yaml [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+                template: '@SyliusAdmin/payment_request/show/content/sections/general/action.html.twig'
+            state:
+                template: '@SyliusAdmin/payment_request/show/content/sections/general/state.html.twig'
+            created_at:
+                template: '@SyliusAdmin/payment_request/show/content/sections/general/created_at.html.twig'
+            updated_at:
+                template: '@SyliusAdmin/payment_request/show/content/sections/general/updated_at.html.twig'
+            
+                
+
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Resources/config/app/twig_hooks/payment_request/show.yaml]:
```suggestion
```

File: src/Sylius/Bundle/AdminBundle/Resources/config/grids/payment.yml [hunk 14]
```diff
@@ -69,6 +69,15 @@ sylius_grid:
                         class: "%sylius.model.channel.class%"
             actions:
                 item:
+                    show_payment_requests:
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Resources/config/grids/payment.yml]:
```suggestion
                    list_payment_requests:
```
🤔

**Review by GSadee — APPROVED** [review]
But the rebase is necessary

**Review by GSadee — APPROVED** [review]

**Pull request merged**